panic: runtime error: invalid memory address or nil pointer dereference after upgrade from RELEASE.2024-02-04T22-36-13Z to RELEASE.2024-03-15T01-07-19Z

## NOTE
After the upgrade from `RELEASE.2024-02-04T22-36-13Z` to `RELEASE.2024-03-15T01-07-19Z` I encounter regular crashes due to an invalid memory access.
Looks like the problem is in the healthCheck trying to reconnect to a peer.

## Expected Behaviour
MinIO does not crash when trying to reconnect to a peer.

## Current Behaviour
MinIO crashes with the following stack trace:
```
Error: *fmt.wrapError
       7: internal/logger/logonce.go:118:logger.(*logOnceType).logOnceIf()
       6: internal/logger/logonce.go:149:logger.LogOnceIf()
       5: internal/rest/client.go:319:rest.(*Client).Call()
       4: cmd/storage-rest-client.go:192:cmd.(*storageRESTClient).call()
       3: cmd/storage-rest-client.go:538:cmd.(*storageRESTClient).ReadVersion()
       2: cmd/erasure-metadata-utils.go:167:cmd.readAllFileInfo.func1()
       1: github.com/minio/pkg/v2@v2.0.11/sync/errgroup/errgroup.go:123:errgroup.(*Group).Go.func1()
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc61cd4]

goroutine 2482070438 [running]:
github.com/minio/minio/internal/logger.buildLogEntry({0x4d4c458?, 0x65277c0?}, {0xc3ede48c80, 0x7b}, {0x0, 0x0, _}, {0xc2720afdf0, 0x1, 0x1})
	github.com/minio/minio/internal/logger/logger.go:367 +0x9b4
github.com/minio/minio/internal/logger.logToEntry(...)
	github.com/minio/minio/internal/logger/logger.go:288
github.com/minio/minio/internal/logger.Event({0x4d4c458, 0x65277c0}, {0x2d76b16?, 0x58b5348?}, {0xc2720aff80?, 0x64e5f80?, 0xc3c2ca1880?})
	github.com/minio/minio/internal/logger/logger.go:422 +0xdb
github.com/minio/minio/internal/rest.(*Client).runHealthCheck.func1()
	github.com/minio/minio/internal/rest/client.go:469 +0x2fa
created by github.com/minio/minio/internal/rest.(*Client).runHealthCheck in goroutine 2482070402
	github.com/minio/minio/internal/rest/client.go:453 +0x6b
```

## Possible Solution
n/a

## Steps to Reproduce (for bugs)
n/a

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Regression
<!-- Is this issue a regression? (Yes / No) -->
<!-- If Yes, optionally please include minio version or commit id or PR# that caused this regression, if you have these details. -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used (`minio --version`): `quay.io/minio/minio:RELEASE.2024-03-15T01-07-19Z`
* Server setup and configuration:
```
  MINIO_BROWSER: "on"
  MINIO_BROWSER_LOGIN_ANIMATION: "off"
  MINIO_DOMAIN: XXX
  MINIO_ERASURE_SET_DRIVE_COUNT: "8"
  MINIO_KMS_KES_CAPATH: /certs/CAs/ca-certificates.crt
  MINIO_KMS_KES_CERT_FILE: /certs/client.crt
  MINIO_KMS_KES_ENDPOINT: https://kes.minio.svc:7373
  MINIO_KMS_KES_KEY_FILE: /certs/client.key
  MINIO_KMS_KES_KEY_NAME: default
  MINIO_NOTIFY_KAFKA_BROKERS_PRIMARY: XXX:9094
  MINIO_NOTIFY_KAFKA_ENABLE_PRIMARY: "on"
  MINIO_NOTIFY_KAFKA_QUEUE_DIR: /events
  MINIO_NOTIFY_KAFKA_SASL_MECHANISM_PRIMARY: sha512
  MINIO_NOTIFY_KAFKA_SASL_PRIMARY: "on"
  MINIO_NOTIFY_KAFKA_TLS_CLIENT_AUTH_PRIMARY: "0"
  MINIO_NOTIFY_KAFKA_TLS_PRIMARY: "on"
  MINIO_NOTIFY_KAFKA_TLS_SKIP_VERIFY_PRIMARY: "on"
  MINIO_NOTIFY_KAFKA_TOPIC_PRIMARY: XXX
  MINIO_NOTIFY_KAFKA_VERSION_PRIMARY: 3.4.0
  MINIO_PROMETHEUS_AUTH_TYPE: public
  MINIO_PROMETHEUS_JOB_ID: minio
  MINIO_SERVER_URL: https://minio.minio.svc:443
  MINIO_STORAGE_CLASS_RRS: EC:1
  MINIO_STORAGE_CLASS_STANDARD: EC:3
  MINIO_UPDATE: "off"
  MINIO_VOLUMES: https://pool-ab961d-{0...3}-0.minio-hl.minio.svc:9000/drives/data{0...89}/minio
```
* Operating System and version (`uname -a`): `Linux 5.15.142-flatcar #1 SMP Mon Dec 11 21:37:48 -00 2023 x86_64`

--- a/internal/logger/logger.go
+++ b/internal/logger/logger.go
@@ -364,7 +364,9 @@ func buildLogEntry(ctx context.Context, message string, trace []string, errKind 
 		entry.API.Args.Bucket = HashString(entry.API.Args.Bucket)
 		entry.API.Args.Object = HashString(entry.API.Args.Object)
 		entry.RemoteHost = HashString(entry.RemoteHost)
-		entry.Trace.Variables = make(map[string]interface{})
+		if entry.Trace != nil {
+			entry.Trace.Variables = make(map[string]interface{})
+		}
 	}
 
 	return entry

